File: projects/01/rectangle_area.rb
#! /usr/bin/ruby
# rectangle_area.rb computes the area of a rectangle given its length and width
#
# Input: the length and width of a rectangle
# Precondition: the length and width are not negative
# Output: the area of the rectangle
#
# Author: Ian Christensen
# Date: February 1, 2018

# function rectangleArea returns a rectangle's area, given its length and width
# Parameters: l, a number && w, a number
# Precondition: l > 0 && w > 0
# Returns: the area of a rectangle whose length is l and whose width is w
def rectangleArea(l, w)	# Declare the definition of the area function passing in the length and width
	l * w				# Compute the area by multiplying the length and width
end						# End the function

if __FILE__ == $0								# If the main file is being run than procede
	puts "To compute the are of a rectangle,"	# Output a message followed by a newline char
	print " enter its length: "					# Prompt the user for the length
	l = gets.chomp.to_f							# Store the length as a float
	print " enter its width: "					# Prompt the user for the width
	w = gets.chomp.to_f							# Store the width as a float
	print "The rectangle's area is: "			# Output a message without a newline char
	puts rectangleArea(l, w)					# Output the final result
end												# End the if statement

File: projects/01/RectangleArea.java
/* RectangleArea.java computes the area of a rectangle.
 *
 * Input: The length and width of the rectangle
 * Preconditions: The length and width are positive numbers.
 * Output: The area of the rectangle.
 *
 * Author: Ian Christensen
 * Date: February 1, 2018
 */

import java.util.Scanner;	// include for i/o operations

public class RectangleArea {
		/* function RectangleArea() computes a rectangle's area, given its length and width
		 * Parameter: l, a double
		 *			  w, a double
		 * Precondition: l and w are not negative.
		 * Returns: the area of the rectangle
		 */
		public static double RectangleArea(double l, double w) {	// declare method and pass in the parameters
			return l * w;											// return the calculated area
		}															// end the RectangleArea method
																	// begin the main program
	public static void main(String[] args) {						// declare the method and pass in string args
		System.out.println("To compute the area of a rectangle,");	// print statement with newline char
		Scanner keyboard = new Scanner(System.in);					// create a Scanner object to check for key strokes
		System.out.print(" enter its length: ");					// prompt user for length without newline
		double l = keyboard.nextDouble();							// set the next entered number as the length
		System.out.print(" enter its width: ");						// prompt the user for width without newline
		double w = keyboard.nextDouble();							// set the next entered
		System.out.println("The area is " + RectangleArea(l, w));	// print the output of the RectangleArea function
	}																// end the main function
}																	// end of the class contents
